File: rs2-rent-sistem.models/Models/CartItem.cs
namespace rs2_rent_sistem.Models.Models
{
    public class CartItem
    {
        public int ID { get; set; }
        public int? Quantity { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public virtual Equipment? Equipment { get; set; }
    }
}

File: rs2-rent-sistem/rs2-ui/rs2_rent_sistem/lib/models/cart_item.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:rs2_rent_sistem/models/equipment_list_item.dart';

part 'cart_item.freezed.dart';
part 'cart_item.g.dart';

@Freezed()
class CartItem with _$CartItem {
  const factory CartItem({
    required int id,
    required DateTime startDate,
    required DateTime endDate,
    required EquipmentListItem equipment,
    @Default(0) int quantity,
  }) = _CartItem;

  factory CartItem.fromJson(Map<String, dynamic> json) => _$CartItemFromJson(json);
}
